src/internal/log/http.go

```diff
@@ -44,7 +46,43 @@ func AddLoggerToHTTPServer(rctx context.Context, name string, s *http.Server) {
 			ctx = ChildLogger(ctx, "", WithFields(id))
 			ctx = metadata.NewOutgoingContext(ctx, metadata.MD{"x-request-id": requestID})
 			r = r.WithContext(ctx)
+
+			var respStatusCode int
+			var errbuf []byte
+			w = httpsnoop.Wrap(w, httpsnoop.Hooks{
+				WriteHeader: func(next httpsnoop.WriteHeaderFunc) httpsnoop.WriteHeaderFunc {
+					return func(code int) {
+						respStatusCode = code
+						next(code)
+					}
+				},
+				Write: func(next httpsnoop.WriteFunc) httpsnoop.WriteFunc {
+					return func(b []byte) (int, error) {
+						if respStatusCode == 0 {
+							respStatusCode = 200
+						}
+						if respStatusCode >= 400 {
```

Comment:
I don't think it would be terrible to always do this, even on successful requests.